File: count_positives_sum_negatives.py
#  https://www.codewars.com/kata/576bb71bbbcf0951d5000044/train/python - 8 kyu

# first method uses a conditional that basically asks if arr exists -> can be replaced with if not arr:
def count_positives_sum_negatives(arr):
    if arr == None or len(arr) == 0:
        return []
    output = []
    posCnt = 0
    negSum = 0
    for i in range(len(arr)):
        if arr[i] > 0:
            posCnt = posCnt + 1
        if arr[i] < 0:
            negSum = negSum + arr[i]
    output.append(posCnt)
    output.append(negSum)

    return output


def count_positives_sum_negatives2(arr):
    if not arr:
        return []
    pos_cnt = 0
    neg_sum = 0
    for x in arr:
        if x > 0:
            pos_cnt += 1
        if x < 0:
            neg_sum += x
    return [pos_cnt, neg_sum]

File: countPositivesSumNegatives.js
// https://www.codewars.com/kata/576bb71bbbcf0951d5000044/train/javascript - 8 kyu

function countPositivesSumNegatives(input) {
  if (input === null || input.length === 0) return [];
  let outputArr = [];
  let countPos = 0;
  let sumNeg = 0;
  for (let i = 0; i < input.length; i++) {
    if (input[i] > 0) {
      countPos += 1;
    }
    if (input[i] < 0) {
      sumNeg += input[i];
    }
  }
  outputArr.push(countPos);
  outputArr.push(sumNeg);
  return outputArr;
}

const countPositivesSumNegatives2 = (input) => {
  if (input === null || input.length === 0) return [];
  let outputArr = [];
  let positives = input.filter((el) => el > 0);
  let negatives = input.filter((el) => el < 0);
  outputArr.push(positives.length);
  outputArr.push(negatives.reduce((a, b) => a + b, 0));
  return outputArr;
};

// one - liner
const countPositivesSumNegatives3 = (input) =>
  input && input.length
    ? [
        input.filter((el) => el > 0).length,
        input.filter((el) => el < 0).reduce((a, b) => a + b, 0),
      ]
    : [];
